discovery/eureka.go

```diff
@@ -0,0 +1,44 @@
+package discovery
+
+import (
+	"errors"
+	"net/http"
+)
+
+//TODO: This should become a discovery interface. And Eureka just the first implementation
+type EurekaClient struct {
+	eurekaUrl string
+}
+
+var errNoEurekaConnection = errors.New("Unable to reach Eureka server")
+var errEurekaUnexpectedHttpResponseCode = errors.New("Eureka returned a non 200 http response code")
+
+func NewEurekaClient(eurekaUrl string) (ec EurekaClient, err error) {
+	ec.eurekaUrl = eurekaUrl
+	resp, err := http.Get(eurekaUrl)
+	if err != nil {
+		return ec, errNoEurekaConnection
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		return ec,errEurekaUnexpectedHttpResponseCode
+	}
+	return ec, nil
+}
+
+var errNoIpsFound = errors.New("No IPs associated to the requested App name")
```

Comment:
Just as a comment, is it a best practice to declare the errors just above the function that will use them? what happens when you want to share error among several functions?

Reply:
The best practice is to declare what you need as close as possible from your need. If you use something several times... just put this where its clearer ;)